src/api/app/models/workflow/step/link_package_step.rb

```diff
@@ -0,0 +1,107 @@
+class Workflow::Step::LinkPackageStep < ::Workflow::Step
+  validates :source_package_name, presence: true
+
+  def call(options = {})
+    return unless valid?
+
+    workflow_filters = options.fetch(:workflow_filters, {})
+
+    # Updated PR
+    if validator.updated_pull_request? && target_package.present?
+      linked_package = target_package
+      update_subscriptions(linked_package, workflow_filters)
+    else # New PR
+      linked_package = create_linked_package
+      create_subscriptions(linked_package, workflow_filters)
+    end
+
+    add_or_update_branch_request_file(package: linked_package)
+    report_to_scm(workflow_filters)
+    linked_package
+  end
+
+  private
+
+  def target_project
+    Project.find_by(name: target_project_name)
+  end
+
+  def source_package
+    Package.find_by_project_and_name(source_project_name, source_package_name)
+  end
+
+  def special_package_file_content
+    <<~XML
+      <services>
+        <service name="format_spec_file" mode="localonly"/>
+      </services>
+    XML
+  end
+
+  def create_project_and_package
+    check_source_access
+
+    if target_package.present?
+      raise Workflow::Step::Errors::DuplicatePackageError.new(target_project_name, target_package_name), "Can not link package. The package #{target_package_name} already exists."
+    end
+
+    if target_project.nil?
+      project = Project.create!(name: target_project_name)
+      project.commit_user = User.session
+      project.relationships.create!(user: User.session, role: Role.find_by_title('maintainer'))
+      project.store
+    end
+
+    target_project.packages.create(name: target_package_name)
+  end
+
+  # NOTE: the next lines are a temporary fix the allow the service to run in a linked package. A project service is needed.
+  def create_special_package
+    return if Package.find_by_project_and_name(target_project, '_project')
+
+    special_package = target_project.packages.create(name: '_project')
+    special_package.save_file({ file: special_package_file_content, filename: '_service' })
+  end
+
+  def create_linked_package
+    create_project_and_package
+    create_special_package
+    link
+  end
+
+  def check_source_access
+    return if remote_source?
+
+    options = { use_source: false, follow_project_links: true }
+    src_package = Package.get_by_project_and_name(source_project_name, source_package_name, options)
+
+    raise Package::Errors::UnknownObjectError if src_package.nil?
```

Comment:
Hm why? If it's nil we know it's "below" a valid remote project. The assumption then is you have access.

Reply:
Done.